test/rules_pygen/rules_generator_test.py

```diff
@@ -0,0 +1,88 @@
+import operator
+import unittest
+import unittest.mock
+
+
+class WhenGeneratingBuildFilesTest(unittest.TestCase):
+    def test_that_wheel_compatibility_is_correct(self):
+        from rules_pygen.rules_generator import _check_compatibility
+
+        self.assertTrue(_check_compatibility('pytest-3.2.3-py2.py3-none-any.whl'))
+        self.assertTrue(_check_compatibility('pytest_hidecaptured-0.1.2-py3-none-any.whl'))
+        self.assertTrue(_check_compatibility('Paste-2.0.3-py34-none-any.whl'))
+        self.assertTrue(_check_compatibility('cryptography-2.3.1-cp34-abi3-macosx_10_6_intel.whl'))
+        self.assertTrue(_check_compatibility('cryptography-2.3.1-cp34-abi3-manylinux1_x86_64.whl'))
+
+        self.assertFalse(_check_compatibility('cryptography-1.6-cp35-cp35m-linux_x86_64.whl'))
+        self.assertFalse(_check_compatibility('cryptography-0.7.1-cp27-none-linux_x86_64.whl'))
+        self.assertFalse(_check_compatibility('tornado-5.1.1-cp27-cp27mu-linux_x86_64.whl'))
+
+    @unittest.mock.patch('rules_pygen.rules_generator._calc_sha256sum')
+    def test_that_wheels_can_be_compared(self, mock_checksum):
+        from rules_pygen.rules_generator import WheelInfo
+
+        wi1 = WheelInfo(
+            '/path/to/foo-1.4.5-py3-none-linux_x68_64.whl', 'https://example.org', 'foo', '1.4.5'
+        )
+        wi2 = WheelInfo(
+            '/path/to/foo-1.4.5-py3-none-macosx_10_6_intel.whl',
+            'https://example.org',
+            'foo',
+            '1.4.5',
+        )
+
+        wheels = [wi1, wi2]
+        wheels.sort(key=operator.attrgetter('platform'))
+        self.assertTrue(wheels[0].filename == 'foo-1.4.5-py3-none-linux_x68_64.whl')
+
+    @unittest.mock.patch('rules_pygen.rules_generator._calc_sha256sum')
+    def test_that_we_dont_add_multiple_purelib_wheels(self, mock_checksum):
+        from rules_pygen.rules_generator import WheelInfo, DependencyInfo
+
+        di = DependencyInfo('foo', [], [])
+        wi1 = WheelInfo(
+            '/path/to/foo-1.4.5-py3-none-linux_x68_64.whl', 'https://example.org', 'foo', '1.4.5'
+        )
+        wi2 = WheelInfo(
+            '/path/to/foo-1.4.5-py3-none-macosx_10_6_intel.whl',
+            'https://example.org',
+            'foo',
+            '1.4.5',
+        )
+        wi3 = WheelInfo(
+            '/path/to/foo-1.4.5-py3-none-manylinux1_x86_64.whl',
+            'https://example.org',
+            'foo',
+            '1.4.5',
+        )
+        wi4 = WheelInfo(
+            '/path/to/foo-1.4.5-py3-none-any.whl', 'https://example.org', 'foo', '1.4.5'
+        )
+
+        di.add_wheel(wi1)
+        self.assertTrue(len(di.wheels) == 1)
+
+        di.add_wheel(wi2)
+        self.assertTrue(len(di.wheels) == 2)
+
+        di.add_wheel(wi3)  # should be ignored because we already have that platform
+        self.assertTrue(len(di.wheels) == 2)
```

Comment:
Might be better if instead of the length you were checking the actual contents of `di.wheels`

Reply:
checking filename as well